Mishmash-Backend/main.js

```diff
@@ -0,0 +1,112 @@
+import { v4 as uuidv4 } from 'uuid'
+import express from 'express'
+import cors from 'cors'
+
+const app = express()
+
+app.use(cors())
+app.use(express.json())
+
+let listIngredients = []
+let listRecipes = []
+
+app.route('/api/ingredient')
+  .get((req, res) => {
+    res.json(listIngredients)
+  })
+  .post((req, res) => {
+    const uuid = uuidv4()
+    const ingredientName = req.body.ingredientName
+
+    if (ingredientName === '') {
+      res.status(404).end()
+    }
+
+    listIngredients = listIngredients.concat([{
+      ingredientUUID: uuid,
+      ingredientName: ingredientName
+    }])
+
+    res.json({
+      ingredientUUID: uuid,
+      ingredientName: ingredientName
+    })
+  })
+  .put((req, res) => {
+    const id = req.body.ingredientUUID
+    const ingredientName = req.body.ingredientName
+    
+    const renamed = [listIngredients.find(value => value.ingredientUUID === id)]
```

Comment:
I don't get it.
`find` will return `undefined` or `ingredient`,
so if below makes sense, but why would you wrap it in the array and filter if you can check it directly: